lxc/completion.go

```diff
@@ -372,8 +372,11 @@ func (g *cmdGlobal) cmpInstanceSetKeys(instanceName string) ([]string, cobra.She
 
 	for configKey := range instance.Config {
 		// We only want to return the intersection between allInstanceConfigKeys and configKeys to avoid returning the full instance config.
+		// We also need to account for user defined keys such as user.* and linux.sysctl.*.
 		_, exists := keySet[configKey]
-		if exists {
+		if exists || strings.HasPrefix(configKey, "user.") ||
+			strings.HasPrefix(configKey, "linux.sysctl.") ||
+			strings.HasPrefix(configKey, "environment.") {
```

Comment:
I just realized I pointed you to the wrong `shared.` util. Here's what I meant:
```suggestion
		if exists || shared.StringHasPrefix(configKey, "user.", "linux.sysctl.", "environment.") {
```

Reply:
Gotcha - this one will work well :)